lib/cylc/scheduler.py

```diff
@@ -1331,20 +1340,155 @@ def suite_shutdown(self):
             self.command_kill_tasks()
             self.time_next_kill = time() + self.INTERVAL_STOP_KILL
 
+        # Is the suite set to auto restart now?
+        if (self.auto_restart_time is not None and
+                time() > self.auto_restart_time):
+            self.auto_restart = True
+            self._set_stop(TaskPool.STOP_REQUEST_NOW_NOW)
+
+    def suite_auto_restart(self, max_retries=3):
+        """Attempt to restart the suite assuming it has already stopped."""
+        cmd = ['cylc', 'restart', quote(self.suite)]
+
+        for attempt_no in range(max_retries):
+            new_host = cylc.scheduler_cli.HostAppointer(
+                cashed=True).appoint_host()
+            LOG.info('Attempting to restart on "%s"' % new_host)
+
+            # proc will start with current env (incl CYLC_HOME etc)
+            proc = Popen(
+                cmd + ['--host=%s' % new_host],
```

Comment:
Would a `cylc restart` without `--host=HOST` work here? In which case, we would no longer need the `HostAppointer`.

Reply:
Yes, we could do it that way, the reasons I went down this route are:

1. We can make sure that there is an available host to restart on before shutting down the suite.
2. We can log the host the suite will attempt to restart on.